### PR TITLE
[move-prover] Verified the permission-related governance rules

### PR DESCRIPTION
**Pull request by junkil-park**
- added/verified the formal specs from the permission aspects of the governance rules
    - some specs are preliminary, which will be improved later.

- TODO: the permissions UpdateValidatorConfig and SetValidatorOperator were not covered yet.

## Motivation

To verify the standard contracts against the governance rules 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

cargo test

### CODE REVIEW

File: language/stdlib/modules/Libra.move [hunk 1]
```diff
@@ -1004,6 +1010,26 @@ module Libra {
     /// aggregate value exceeds the CoinInfo.total_value.  However, that property involves summations over
     /// all resources and is beyond the capabilities of the specification logic or the prover, currently.
 
+    // TODO: This schema is better to be in Roles, but put here to avoid an unused schema warning from Roles.
+    spec schema AbortsIfNotDesignatedDealer {
+        account: signer;
+        aborts_if !Roles::spec_has_designated_dealer_role_addr(Signer::spec_address_of(account));
+    }
+
+    spec module {
```
[the remark this answers is not in the document]
Comment (DavidLDill) [reply language/stdlib/modules/Libra.move:1019]: To be clear, I don't think this requires action.

File: language/stdlib/modules/DualAttestation.move [hunk 2]
```diff
@@ -91,6 +91,14 @@ module DualAttestation {
             expiration_date: U64_MAX,
         })
     }
+    spec fun publish_credential {
+        pragma aborts_if_is_partial = true;
+
+        // TODO(jkpark): Refactor this governance spec.
+        /// The permission "RotateDualAttestationInfo" is granted to ParentVASP, DesignatedDealer [B26].
+        aborts_if !Roles::spec_has_parent_VASP_role_addr(Signer::spec_address_of(created)) &&
```
[the remark this answers is not in the document]
Comment (junkil-park) [reply language/stdlib/modules/DualAttestation.move:99]: This spec has the form of "A && B", so I think it's difficult to split it into two "aborts_if" specs.

File: language/stdlib/modules/Libra.move (hunk 1 above)
[the remark this answers is not in the document]
Comment (junkil-park) [reply language/stdlib/modules/Libra.move:1019]: I strongly agree that we'd like stronger formal properties regarding "permissions".

**Pull request closed without merging**